lib/matplotlib/legend.py

```diff
@@ -1319,10 +1319,8 @@ def _parse_legend_args(axs, *args, handles=None, labels=None, **kwargs):
     handlers = kwargs.get('handler_map')
 
     if (handles is not None or labels is not None) and args:
-        _api.warn_deprecated("3.9", message=(
-            "You have mixed positional and keyword arguments, some input may "
-            "be discarded.  This is deprecated since %(since)s and will "
-            "become an error in %(removal)s."))
+        raise ValueError("When passing handles and labels, they must both be "
+                         "passed positionally or both as keywords.")
```

Comment:
Is there a better exception type than `ValueError` for this situation?

Reply:
There's no really good match. I'm inclined to go with TypeError as that is used in other cases, when the signature is not fulfilled.